test/e2e-test/app_resourcetracker_test.go

```diff
@@ -138,19 +126,28 @@ var _ = Describe("Test application cross namespace resource", func() {
 		Expect(k8sClient.Delete(ctx, app)).Should(Succeed())
 		By("Verify cluster-scoped trait is deleted cascadingly")
 		Eventually(func() error {
-			return k8sClient.Get(ctx, client.ObjectKey{Name: "pv-" + componentName, Namespace: namespace}, pv)
-		}, 20*time.Second, 500*time.Millisecond).Should(SatisfyAll(&util.NotFoundMatcher{}))
+			if err := k8sClient.Get(ctx, client.ObjectKey{Name: "pv-" + componentName, Namespace: namespace}, pv); err != nil {
+				if apierrors.IsNotFound(err) {
+					return nil
+				}
+				return errors.Wrap(err, "PersistentVolume has not deleted")
+			}
+			if ctrlutil.ContainsFinalizer(pv, "kubernetes.io/pv-protection") {
+				return nil
```

Comment:
why it means deleted?

Reply:
This PV is not bound to anything, so if this finalizer is found, it's safe to assert it will be deleted.
But the GC controller may take a long time to gc it. So I just use this to check it's deleted cascadingly.